sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2086,6 +2087,20 @@ class Analyzer(
     }
   }
 
+  /**
+   * Set the seed for random number generation in Shuffle expressions.
+   */
+  object ResolvedShuffleExpressions extends Rule[LogicalPlan] {
+    private lazy val random = new Random()
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.transformUp {
+      case p if p.resolved => p
+      case p => p transformExpressionsUp {
+        case Shuffle(child, None) => Shuffle(child, Some(random.nextLong()))
```

Comment:
this looks reasonable, do we have any context about why we start doing this?

Reply:
I followed the implementation of `Uuid` and when we started doing this was at #20861. cc @viirya

Reply:
then can we use a single rule to assign seed to these randomized functions?

Reply:
Yeah, in `Uuid` we want to make sure the same query plan can return the same result. It is more deterministic between retries.